homeassistant/components/intesishome/climate.py

```diff
@@ -0,0 +1,390 @@
+"""
+Support for IntesisHome Smart AC Controllers.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/intesishome/
+"""
+import asyncio
+import logging
+from datetime import timedelta
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+
+from homeassistant.components import persistent_notification
+from homeassistant.components.climate import PLATFORM_SCHEMA, ClimateDevice
+from homeassistant.components.climate.const import (ATTR_OPERATION_MODE,
+                                                    STATE_AUTO, STATE_COOL,
+                                                    STATE_DRY, STATE_FAN_ONLY,
+                                                    STATE_HEAT,
+                                                    SUPPORT_FAN_MODE,
+                                                    SUPPORT_OPERATION_MODE,
+                                                    SUPPORT_SWING_MODE,
+                                                    SUPPORT_TARGET_TEMPERATURE)
+from homeassistant.const import (ATTR_TEMPERATURE, CONF_PASSWORD,
+                                 CONF_USERNAME, STATE_OFF, STATE_UNKNOWN,
+                                 TEMP_CELSIUS)
+from homeassistant.exceptions import PlatformNotReady
+
+REQUIREMENTS = ['pyintesishome==0.8']
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string
+})
+
+# Return cached results if last scan time was less than this value.
+# If a persistent connection is established for the controller, changes to
+# values are in realtime.
+SCAN_INTERVAL = timedelta(seconds=300)
+
+IH_FAN_AUTO = 'Auto'
+IH_FAN_QUIET = 'Quiet'
+IH_FAN_LOW = 'Low'
+IH_FAN_MEDIUM = 'Medium'
+IH_FAN_HIGH = 'High'
+
+IH_SWING_WIDGET = 42
+IH_SWING_STOP = 'Auto/Stop'
+IH_SWING_BOTH = 'Both'
+IH_SWING_HORIZONTAL = 'Horizontal'
+IH_SWING_VERTICAL = 'Vertical'
+
+MAP_OPERATION_MODE = {
+    'auto': STATE_AUTO,
+    'fan': STATE_FAN_ONLY,
+    'heat': STATE_HEAT,
+    'dry': STATE_DRY,
+    'cool': STATE_COOL,
+    'off': STATE_OFF
+}
+
+MAP_STATE_ICONS = {
+    STATE_HEAT: 'mdi:white-balance-sunny',
+    STATE_AUTO: 'mdi:cached',
+    STATE_COOL: 'mdi:snowflake',
+    STATE_DRY: 'mdi:water-off',
+    STATE_FAN_ONLY: 'mdi:fan',
+}
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Create the IntesisHome climate devices."""
+    from pyintesishome import IntesisHome
+
+    ihuser = config[CONF_USERNAME]
+    ihpass = config[CONF_PASSWORD]
+
+    controller = IntesisHome(ihuser, ihpass, hass.loop)
+
+    await hass.async_add_executor_job(controller.connect)
+    await asyncio.sleep(2)
```

Comment:
really necessary ? what is going on for you to need this sleep ?

Reply:
The pyintesishome library doesn't yet have a callback for connection made, so it needs to sleep for a moment to wait for the device list to be populated. Connect method only initiates the connection so wasn't sure how else to do it other than looping on is_connected